javalib/src/main/scala/java/io/InputStream.scala

```diff
@@ -58,6 +58,17 @@ abstract class InputStream extends Closeable {
     readNBytes(Integer.MAX_VALUE)
   }
 
+  private def readNBytesImpl(buffer: Array[Byte], off: Int, len: Int): Int = {
+    var totalBytesRead = 0
+
+    while (totalBytesRead < len && {
```

Comment:
As a person who reads this code and may someday have to maintain it, I ask if this code _has_ to be so dense?

Yes it is concise and one can discuss if it is idiomatic
until the cows come home.   

There are academic measures of complexity for code.
I bet this block scores pretty high.

I believe the aiming point should be a competent
second year graduate student or a new hire with Scala experience, not an APL programmer.

I'd rather be off chasing fatal bugs in the current code base than taking the time to parse stuff.